manifests/deploy.pp

```diff
@@ -27,6 +27,7 @@
   $servergroups = hiera('jboss::deploy::servergroups', undef),
   $controller   = $::jboss::controller,
   $runasdomain  = $::jboss::runasdomain,
+  $runtime_name = undef,
```

Comment:
As you can see in `tests/deploy.pp`, the variable `$runtime_name` is *not* set and it works nevertheless:
```
jboss::deploy { "${artifact_name}.war":
  ensure      => 'present',
  redeploy    => 'true',
  path        => "/usr/src/${artifact_fullname}.war",
}
```
Further, the following snippet in `jboss_deploy/jbosscli.rb` ignores the parameter if it evaluates to nil:
```
  def runtime_name_param
    if @resource[:runtime_name].nil?
      return ''
    else
      return "--runtime-name=#{@resource[:runtime_name]}"
    end
  end
```